### PR TITLE
br: unified the lint rules with tidb.

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31643

Problem Summary: unified the golang-lint rules with tidb.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/tidb/tidb_test.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func TestWriteRowsErrorOnDup(t *testing.T) {
 }
 
 // TODO: temporarily disable this test before we fix strict mode
-//nolint:unused
+//nolint
```
Comment (kennytm) [inline br/pkg/lightning/backend/tidb/tidb_test.go:217]: what other lints trigger this?

File: br/pkg/restore/util_test.go [hunk 2]
```diff
@@ -224,8 +224,10 @@ func TestPaginateScanRegion(t *testing.T) {
 
 	ctx := context.Background()
 	regionMap := make(map[uint64]*restore.RegionInfo)
-	regions := []*restore.RegionInfo{}
-	batch, err := restore.PaginateScanRegion(ctx, NewTestClient(stores, regionMap, 0), []byte{}, []byte{}, 3)
+	var regions []*restore.RegionInfo
+	var batch []*restore.RegionInfo
+	var err error
+	_, err = restore.PaginateScanRegion(ctx, NewTestClient(stores, regionMap, 0), []byte{}, []byte{}, 3)
```
Comment (kennytm) [inline br/pkg/restore/util_test.go:230]:
```suggestion
	_, err := restore.PaginateScanRegion(ctx, NewTestClient(stores, regionMap, 0), []byte{}, []byte{}, 3)
```

**Review by kennytm — COMMENTED** [review]
rest lgtm

**Review by kennytm — APPROVED** [review]

File: Makefile.common [hunk 3]
```diff
@@ -46,13 +46,15 @@ LINUX     := "Linux"
 MAC       := "Darwin"
 
 PACKAGE_LIST  := go list ./...
-PACKAGE_LIST_TIDB_TESTS  := go list ./... | grep -vE "github.com\/pingcap\/tidb\/br|github.com\/pingcap\/tidb\/cmd|github.com\/pingcap\/tidb\/dumpling"
+PACKAGE_LIST_TIDB_TESTS  := go list ./... | grep -vE "github.com\/pingcap\/tidb\/cmd|github.com\/pingcap\/tidb\/dumpling"
```
Comment (tangenta) [inline Makefile.common:49]: `PACKAGES_TIDB_TESTS` is changed, please check all the usages and make sure the semantic is unchanged.

**Review by tangenta — APPROVED** [review]

**Pull request merged**